umbral/fragments.py

```diff
@@ -75,12 +75,79 @@ def __bytes__(self):
         return self.to_bytes()
 
 
+class CorrectnessProof(object):
+    def __init__(self, e2, v2, u1, u2, z1, z2, z3, metadata:bytes=None):
+        self.point_eph_e2 = e2
+        self.point_eph_v2 = v2
+        self.point_kfrag_commitment = u1
+        self.point_kfrag_pok = u2
+        self.bn_kfrag_sig1 = z1
+        self.bn_kfrag_sig2 = z2
+        self.bn_sig = z3
+        self.metadata = metadata
+
+    @classmethod
+    def from_bytes(cls, data: bytes, curve: ec.EllipticCurve=None):
+        """
+        Instantiate CorrectnessProof from serialized data.
+        """
+        curve = curve if curve is not None else default_curve()
+        key_size = get_curve_keysize_bytes(curve)
+        data = BytesIO(data)
+
+        # BigNums are the keysize in bytes, Points are compressed and the
+        # keysize + 1 bytes long.
+        e2 = Point.from_bytes(data.read(key_size + 1), curve)
+        v2 = Point.from_bytes(data.read(key_size + 1), curve)
+        kfrag_commitment = Point.from_bytes(data.read(key_size + 1), curve)
+        kfrag_pok = Point.from_bytes(data.read(key_size + 1), curve)
+        kfrag_sig1 = BigNum.from_bytes(data.read(key_size), curve)
+        kfrag_sig2 = BigNum.from_bytes(data.read(key_size), curve)
+        sig = BigNum.from_bytes(data.read(key_size), curve)
+
+        metadata = data.read()
+        if metadata == bytes(0):
```

Comment:
This is build a bit weirdly.

You can simply do `metadata = data.read() or None`. If there is no data, it will be `b''` which loses precedence to `None`, then you don't have to set it to `None`.

That is unless you are setting `metadata` to be 0x00 and it will always have that byte, but I don't like that.

Reply:
Fixed